src/Testcontainers/Configurations/Modules/Databases/CosmosDbTestcontainerConfiguration.cs

```diff
@@ -0,0 +1,66 @@
+namespace DotNet.Testcontainers.Configurations
+{
+  using System;
+  using System.Globalization;
+  using System.IO;
+  using DotNet.Testcontainers.Builders;
+  using JetBrains.Annotations;
+
+  [PublicAPI]
+  public class CosmosDbTestcontainerConfiguration : TestcontainerDatabaseConfiguration
+  {
+    public const string DefaultCosmosDbApiImage =
+      "mcr.microsoft.com/cosmosdb/linux/azure-cosmos-emulator";
+
+    private const int DefaultCosmosPort = 8081;
+
+    public CosmosDbTestcontainerConfiguration()
+      : this(DefaultCosmosDbApiImage)
+    {
+      this.OutputConsumer = Consume.RedirectStdoutAndStderrToStream(new MemoryStream(), new MemoryStream());
```

Comment:
Move the `OutputConsumer` initialization to the constructor below. Otherwise, it is not always initialized. I think we can make the class `sealed` and use following to initialize the properties:

```csharp
this.PartitionCount = 1;
this.IpAddressOverride = "127.0.0.1";
this.OutputConsumer = Consume.RedirectStdoutAndStderrToStream(new MemoryStream(), new MemoryStream());
this.WaitStrategy = Wait.ForUnixContainer().UntilMessageIsLogged(this.OutputConsumer.Stdout, "Started|Shutting");
```

Reply:
Good point. WaitStrategy is not settable so I kept the override.

Reply:
Yes - you need to override `WaitStrategy`, but you can still set the value in the constructor (read-only; just to create the instance once).